profiling-ffi/src/profiles.rs

```diff
@@ -325,12 +325,36 @@ pub unsafe extern "C" fn ddog_prof_Profile_drop(
 ) {
 }
 
+#[repr(C)]
+pub enum ProfileAddResult {
+    Ok(u64),
+    Err(ddcommon_ffi::Vec<u8>),
+}
+
+#[cfg(test)]
+impl From<ProfileAddResult> for Result<u64, String> {
+    fn from(result: ProfileAddResult) -> Self {
+        match result {
+            ProfileAddResult::Ok(ok) => Ok(ok),
+            ProfileAddResult::Err(err) => {
+                // Safety: not generally safe but this is for test code.
+                Err(unsafe { String::from_utf8_unchecked(err.into()) })
+            }
+        }
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn ddog_prof_ProfileAddResult_drop(_result: ProfileAddResult) {}
```

Comment:
Naming: (Similar to other comments above)

```suggestion
pub extern "C" fn ddog_prof_Profile_AddResult_drop(_result: ProfileAddResult) {}
```